### PR TITLE
feat(project): #1287 added project views count

### LINKED ISSUE #1287: Project views
- [ ] Increment counter each time project is view (by anyone)
- [ ] Display on project lists (homepage/my projects)

### PR DESCRIPTION
**Pull request by webkhushboo**
closes #1287 

### Notes

A summary of what was achieved in this PR

- [ ] API: Added function to count the project views in service
- [ ] UI: Display project views on my project and home page

image

### CODE REVIEW

File: web/src/app/core/services/project.service.ts [hunk 1]
```diff
@@ -162,15 +157,19 @@ export class ProjectService {
   }
 
   /**
-   * This function stripes the trailing slash from url
-   * @param url the url address
-   * @return the url after stripe of the trailing slash
+   * Function to update the project views
+   * @param uid uid of project
```
Comment (webkhushboo) [inline web/src/app/core/services/project.service.ts]: We need to comment as per function arguments

File: web/src/app/shared/models/project.model.ts [hunk 2]
```diff
@@ -73,4 +79,75 @@ export class ProjectModel {
...
+
+  /**
+   * Check if owner of the project
+   * @param userUid user uid
+   */
+  public isAdmin(userUid: string): boolean {
+    return this.access.admin.includes(userUid);
+  }
+
+  public toData(): IProject {
```
Comment (webkhushboo) [inline web/src/app/shared/models/project.model.ts]: We should comment here

File: web/src/app/shared/models/factories/model.factory.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { IModel, Model } from '../model.model';
+
+export class ModelFactory {
```
Comment (webkhushboo) [inline web/src/app/shared/models/factories/model.factory.ts]: We should add comment here

File: web/src/app/shared/models/access.model.ts [hunk 4]
```diff
@@ -1,4 +1,24 @@
...
+  admin: string[];
+  readonly?: string[];
+
+  constructor(access?: IAccess) {
+    super();
+    this.admin = access && access.admin ? access.admin : [];
+    this.readonly = access && access.readonly ? access.readonly : [];
+  }
+
+  public toData(): IAccess {
```
Comment (webkhushboo) [inline web/src/app/shared/models/access.model.ts]: Please add comments

File: web/src/app/core/services/project.service.ts (hunk 1 above)
Comment (eddiejaoude) [reply web/src/app/core/services/project.service.ts]: 👍

File: web/src/app/shared/models/project.model.ts (hunk 2 above)
Comment (eddiejaoude) [reply web/src/app/shared/models/project.model.ts]: 👍

File: web/src/app/shared/models/factories/model.factory.ts (hunk 3 above)
Comment (eddiejaoude) [reply web/src/app/shared/models/factories/model.factory.ts]: 👍

File: web/src/app/shared/models/access.model.ts (hunk 4 above)
Comment (eddiejaoude) [reply web/src/app/shared/models/access.model.ts]: 👍

**Comment by eddiejaoude** [comment]
I will fix the failing CI in my next commit 🤓 - its the new rules so its good 👍

**Comment by webkhushboo** [comment]
@eddiejaoude 

Not able to create token . Getting the below error 
image

**Pull request merged**